lib/matplotlib/transforms.py

```diff
@@ -682,6 +682,63 @@ def intersection(bbox1, bbox2):
 class Bbox(BboxBase):
     """
     A mutable bounding box.
+
+    Examples
+    --------
+    The default constructor takes the boundary "points" ``[[xmin, ymin],
+    [xmax, ymax]]``.
+
+        >>> example_box = Bbox([[1, 1], [3, 7]])
+        >>> example_box
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    Alternatively, a Bbox can be created from the flattened array, the
+    so-called "extents" ``(xmin, ymin, xmax, ymax)``
+
+        >>> Bbox.from_extents(1, 1, 3, 7)
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    or from the "bounds" ``(xmin, ymin, width, height)``.
+
+        >>> Bbox.from_bounds(1, 1, 2, 6)
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    The "empty" object for accumulating Bboxs is the null bbox, which has
+    ``(min, max) = (np.inf, -np.inf)`` for each dimension.
+
+        >>> Bbox.null()
+        Bbox([[inf, inf], [-inf, -inf]])
+
+    Adding points to the null bbox will give you the bbox of those points.
+
+        >>> box = Bbox.null()
+        >>> box.update_from_data_xy([[1, 1]])
+        >>> box
+        Bbox([[1.0, 1.0], [1.0, 1.0]])
+        >>> box.update_from_data_xy([[2, 3], [3, 2]], ignore=False)
+        >>> box
+        Bbox([[1.0, 1.0], [3.0, 3.0]])
+
+    Setting ``ignore=True`` is equivalent to starting over from a null bbox.
+
+        >>> box.update_from_data_xy([[1, 1]], ignore=True)
+        >>> box
+        Bbox([[1.0, 1.0], [1.0, 1.0]])
+
+    The null bbox is also the identity for intersections
+
+        >>> Bbox.intersection(example_box, Bbox.null())
+        Bbox([[1.0, 1.0], [3.0, 7.0]])
+
+    except with itself, where it returns the full space.
+
+        >>> Bbox.intersection(Bbox.null(), Bbox.null())
+        Bbox([[-inf, -inf], [inf, inf]])
+
```

Comment:
Is this a reasonable behavior? Seems to contradict the semantics of update_from_data_xy.

Reply:
Yeah part of why I'm documenting this is because this behavior was very unexpected. I expected `Bbox.null()` to be the "zero" object of the intersection/union ring over the Bbox's. 

But it's not technically a *bug* per se since it's not documented, so I didn't open an issue.

Reply:
That said, I'd be happy to make a separate PR to fix this bug, if other people also agree that it shouldn't be this way, but don't want to get bogged down talking about deprecation timelines, etc. here, would rather open a separate issue.

Reply:
This and below clearly look like bugs, but I'm fine documenting these here and changing them later.